File: clojure/src/spoonacular_api/specs/inline_response_200_6_recipes.clj
(ns spoonacular-api.specs.inline-response-200-6-recipes
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs.inline-response-200-3-extended-ingredients :refer :all]
            [spoonacular-api.specs.inline-response-200-3-wine-pairing :refer :all]
            )
  (:import (java.io File)))


(def inline-response-200-6-recipes-data
  {
   (ds/req :id) int?
   (ds/req :title) string?
   (ds/req :image) string?
   (ds/req :imageType) string?
   (ds/req :servings) float?
   (ds/req :readyInMinutes) int?
   (ds/req :license) string?
   (ds/req :sourceName) string?
   (ds/req :sourceUrl) string?
   (ds/req :spoonacularSourceUrl) string?
   (ds/req :aggregateLikes) float?
   (ds/req :healthScore) float?
   (ds/req :spoonacularScore) float?
   (ds/req :pricePerServing) float?
   (ds/opt :analyzedInstructions) (s/coll-of any?)
   (ds/req :cheap) boolean?
   (ds/req :creditsText) string?
   (ds/opt :cuisines) (s/coll-of string?)
   (ds/req :dairyFree) boolean?
   (ds/opt :diets) (s/coll-of string?)
   (ds/req :gaps) string?
   (ds/req :glutenFree) boolean?
   (ds/req :instructions) string?
   (ds/req :ketogenic) boolean?
   (ds/req :lowFodmap) boolean?
   (ds/opt :occasions) (s/coll-of string?)
   (ds/req :sustainable) boolean?
   (ds/req :vegan) boolean?
   (ds/req :vegetarian) boolean?
   (ds/req :veryHealthy) boolean?
   (ds/req :veryPopular) boolean?
   (ds/req :whole30) boolean?
   (ds/req :weightWatcherSmartPoints) float?
   (ds/opt :dishTypes) (s/coll-of string?)
   (ds/opt :extendedIngredients) (s/coll-of inline-response-200-3-extended-ingredients-spec)
   (ds/req :summary) string?
   (ds/opt :winePairing) inline-response-200-3-wine-pairing-spec
   })

(def inline-response-200-6-recipes-spec
  (ds/spec
    {:name ::inline-response-200-6-recipes
     :spec inline-response-200-6-recipes-data}))

File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse2006Recipes.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse2003ExtendedIngredients
import com.spoonacular.client.model.InlineResponse2003WinePairing
import java.math.BigDecimal
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param title 
 * @param image 
 * @param imageType 
 * @param servings 
 * @param readyInMinutes 
 * @param license 
 * @param sourceName 
 * @param sourceUrl 
 * @param spoonacularSourceUrl 
 * @param aggregateLikes 
 * @param healthScore 
 * @param spoonacularScore 
 * @param pricePerServing 
 * @param analyzedInstructions 
 * @param cheap 
 * @param creditsText 
 * @param cuisines 
 * @param dairyFree 
 * @param diets 
 * @param gaps 
 * @param glutenFree 
 * @param instructions 
 * @param ketogenic 
 * @param lowFodmap 
 * @param occasions 
 * @param sustainable 
 * @param vegan 
 * @param vegetarian 
 * @param veryHealthy 
 * @param veryPopular 
 * @param whole30 
 * @param weightWatcherSmartPoints 
 * @param dishTypes 
 * @param extendedIngredients 
 * @param summary 
 * @param winePairing 
 */
case class InlineResponse2006Recipes(id: Int,
                title: String,
                image: String,
                imageType: String,
                servings: BigDecimal,
                readyInMinutes: Int,
                license: String,
                sourceName: String,
                sourceUrl: String,
                spoonacularSourceUrl: String,
                aggregateLikes: BigDecimal,
                healthScore: BigDecimal,
                spoonacularScore: BigDecimal,
                pricePerServing: BigDecimal,
                analyzedInstructions: Option[Seq[Object]],
                cheap: Boolean,
                creditsText: String,
                cuisines: Option[Seq[String]],
                dairyFree: Boolean,
                diets: Option[Seq[String]],
                gaps: String,
                glutenFree: Boolean,
                instructions: String,
                ketogenic: Boolean,
                lowFodmap: Boolean,
                occasions: Option[Seq[String]],
                sustainable: Boolean,
                vegan: Boolean,
                vegetarian: Boolean,
                veryHealthy: Boolean,
                veryPopular: Boolean,
                whole30: Boolean,
                weightWatcherSmartPoints: BigDecimal,
                dishTypes: Option[Seq[String]],
                extendedIngredients: Option[Seq[InlineResponse2003ExtendedIngredients]],
                summary: String,
                winePairing: Option[InlineResponse2003WinePairing]
                )

object InlineResponse2006Recipes {
    /**
     * Creates the codec for converting InlineResponse2006Recipes from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse2006Recipes] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse2006Recipes] = deriveEncoder
}
